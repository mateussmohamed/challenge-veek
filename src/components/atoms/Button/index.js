import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const baseStyle = css`
  box-shadow: 0 4px 20px -2px rgba(40, 53, 131, 0.34);
  border-radius: 65px;
  min-width: 240px;
  color: #ffffff;
  font-family: Lato, sans-serif;
  font-size: 16px;
  font-weight: 400;
  padding: 6px 10px;
  text-transform: uppercase;
`

const primaryStyle = css`
  background-color: ${props => props.theme.colors.primary};
`

const secondaryStyle = css`
  background-color: ${props => props.theme.colors.secondary};
`

const StyledButton = styled.button`
  ${baseStyle}
  ${props => (props.primary ? primaryStyle : '')}
  ${props => (props.secondary ? secondaryStyle : '')}
`

const Button = props => <StyledButton {...props} />

Button.defaultProps = {
  primary: true,
}

Button.propTypes = {
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
}

export default Button
