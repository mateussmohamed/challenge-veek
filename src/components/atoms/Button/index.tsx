import React from 'react'
import styled, { css } from 'styled-components'

const baseStyle = css`
  min-width: 240px;
  border-radius: 65px;
  padding: 6px 10px;
  border: 0;
  outline: 0;
  color: #ffffff;
  font-family: Lato, sans-serif;
  font-size: 16px;
  font-weight: 400;
  text-transform: uppercase;
  box-shadow: 0 4px 15px -2px rgba(40, 53, 131, 0.4);

  :hover {
    box-shadow: 0 4px 22px -2px rgba(40, 53, 131, 0.7);
  }
`

const primaryStyle = css`
  background-color: ${props => props.theme.colors.primary};
`

const secondaryStyle = css`
  background-color: ${props => props.theme.colors.secondary};
`

interface ButtonProps {
  readonly children: React.ReactNode
  readonly primary?: boolean
  readonly secondary?: boolean
}

const StyledButton = styled.button<ButtonProps>`
  ${baseStyle}
  ${props => props.primary && !props.secondary && primaryStyle}
  ${props => props.secondary && secondaryStyle}

  cursor: pointer;

  & :disabled {
    cursor: default;
    pointer-events: none;
    color: #f1f1f1;
    background-color: #999;
  }
`

export const Button = ({ primary = true, ...restProps }: ButtonProps) => {
  return <StyledButton primary {...restProps} />
}
