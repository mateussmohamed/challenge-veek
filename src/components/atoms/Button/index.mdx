---
name: Button
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import Button from './index'

# Button
<PropsTable of={Button} />

<Playground>
  <Button>Discover us</Button>
  <Button secondary>Validate</Button>
</Playground>
