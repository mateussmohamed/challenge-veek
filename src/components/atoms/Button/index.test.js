import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Button from './index'

describe('[<Button />]', () => {
  const wrap = (props = {}) => shallow(<Button {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap()

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass secondary prop', () => {
    test('matches to snapshot', () => {
      const wrapper = wrap({ secondary: true })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
