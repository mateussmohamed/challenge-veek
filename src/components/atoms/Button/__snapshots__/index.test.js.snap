// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Button />] matches to snapshot 1`] = `
<Button__StyledButton
  primary={true}
/>
`;

exports[`[<Button />] when pass props matches to snapshot 1`] = `
<Button__StyledButton
  primary={true}
  secondary={true}
/>
`;
